File: C++/Manual 03/Question04.cpp
#include <iostream>
using namespace std;

int main()
{
    // Hollow Diamond
    int i, j, k, n;
    char x;
    cout << "Enter number of rows: ";
    cin >> n;
    cout << "Enter the symbol to print with: ";
    cin >> x;

    i = 1;
    while (i <= n)
    {
        k = 1;
        while (k <= n - i)
        {
            cout << " ";
            k++;
        }
        j = 1;
        while (j <= i)
        {
            if (j == 1 || j == i)
            {
                cout << x << " ";
            }
            else
            {
                cout << "  ";
            }
            j++;
        }
        cout << endl;
        i++;
    }
    i = 1;
    while (i <= n)
    {
        k = n;
        while (k > n - i)
        {
            cout << " ";
            k--;
        }
        j = 1;
        while (j <= n - i)
        {
            if (j == 1 || j == (n - i))
            {
                cout << x << " ";
            }
            else
            {
                cout << "  ";
            }
            j++;
        }
        cout << endl;
        i++;
    }
}

File: Python/Manual 03/Question04.py
def print_hollow_diamond(n, x):
    i = 1
    while i <= n:
        k = 1
        while k <= n - i:
            print(" ", end="")
            k += 1
        j = 1
        while j <= i:
            if j == 1 or j == i:
                print(x, end=" ")
            else:
                print(" ", end=" ")
            j += 1
        print()
        i += 1

    i = 1
    while i <= n:
        k = n
        while k > n - i:
            print(" ", end="")
            k -= 1
        j = 1
        while j <= n - i:
            if j == 1 or j == (n - i):
                print(x, end=" ")
            else:
                print(" ", end=" ")
            j += 1
        print()
        i += 1


n = int(input("Enter number of rows: "))
x = input("Enter the symbol to print with: ")
print_hollow_diamond(n, x)
